File: examples/university-demo-OLD/src/main/java/io/axoniq/demo/university/faculty/write/createcourse/CreateCourse.java
package io.axoniq.demo.university.faculty.write.createcourse;

import io.axoniq.demo.university.faculty.write.CourseId;
import org.axonframework.modelling.command.annotation.TargetEntityId;

public record CreateCourse(@TargetEntityId CourseId courseId, String name, int capacity) {

}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/write/create_course_functional/CreateCourse.kt
package org.axonframework.examples.university.faculty.write.create_course_functional

import org.axonframework.examples.university.shared.ids.CourseId
import org.axonframework.modelling.annotation.TargetEntityId

data class CreateCourse(
    val courseId: CourseId,
    val name: String,
    val capacity: Int
) {
    companion object {
        const val ID = "courseId"
    }
}
